widgetssdk/src/main/java/com/glia/widgets/push/notifications/PushClickHandlerController.kt

```diff
@@ -17,53 +20,61 @@ private class PendingPn(val queueId: String?, val visitorId: String)
 internal class PushClickHandlerControllerImpl(
     private val configurationManager: ConfigurationManager,
     private val uiComponentsDispatcher: UiComponentsDispatcher,
-    private val gliaCore: GliaCore
+    private val gliaCore: GliaCore,
+    private val isQueueingOrLiveEngagementUseCase: IsQueueingOrLiveEngagementUseCase,
+    private val isAuthenticatedUseCase: IsAuthenticatedUseCase
 ) : PushClickHandlerController {
 
     // Presence of this object indicates that we're waiting for authentication
     private var pendingPn: PendingPn? = null
 
+    private val intention: Intention
+        get() = when {
+            isQueueingOrLiveEngagementUseCase.hasOngoingLiveEngagement -> Intention.RETURN_TO_CHAT
+            else -> Intention.SC_CHAT
+        }
+
     override fun handlePushClick(queueId: String?, visitorId: String) {
-        pendingPn = PendingPn(queueId, visitorId)
+        if (isAuthenticatedUseCase() || isQueueingOrLiveEngagementUseCase.hasOngoingLiveEngagement) {
+            handlePendingPushNotification(PendingPn(queueId, visitorId))
```

Comment:
Looks a bit suspicious that there is no need to set `pendingPn = PendingPn(queueId, visitorId)` in this condition, only in `else` case. Could you please confirm that the logic is correct here?

Reply:
That is because if visitor is already authenticated or has ongoing live engagement, no need to wait for authentication, therefore we proceed immediately.

Reply:
Got it. Thanks.
That deserves a comment, I think. Might be useful.